firehose.tf

```diff
@@ -0,0 +1,38 @@
+resource "aws_s3_bucket" "firehose_bucket" {
+  count       = var.firehose_enabled ? 1 : 0
+  bucket      = module.this.id
+  acl         = "private"
+}
+
+resource "aws_iam_role" "firehose_role" {
+  count               = var.firehose_enabled ? 1 : 0
+  name                = format("%s_%s", module.this.id, "firehose_aws_fms_role")
```

Comment:
1. We don't use `_` in resource names
2. Use a separate `label` module - see above

```
name                = module.firehose.id
```